MainEvent/main.cpp

```diff
@@ -94,6 +101,51 @@ void MoveRobot() {
   //Roboteq.sendAngles();
 }
 
+bool sortingMethod (Mat& a, Mat& b){
+    return a.Y > b.Y;
+}
+
+void matchingMethod(){
+  // DONE: Write function here to match the balls in the ballCoordL & ballCoordR and put them into vector of pair
+  //sorting here is inefficient(not like it really matters that much here), makes more sense to sort after balls out of bounds removed
+  //sort(ballCoordR.begin(), ballCoordR.end(), sortingMethod());
+  //sort(ballCoordL.begin(), ballCoordL.end(), sortingMethod());
+
+  vector<Mat> correctedR;
+  vector<Mat> correctedL;
+
+  for (auto lball: ballCoordL){
+    if (lball.X > leftBound){
+      correctedL.push_back(lball);
+    }
+  }
+  for (auto rball: ballCoordR){
+    if (rball.X < rightBound){
+      correctedR.push_back(rball);
+    }
+  }
+
+  sort(correctedR.begin(), correctedR.end(), sortingMethod());
+  sort(correctedL.begin(), correctedL.end(), sortingMethod());
+  //balls are now sorted by their Y values
+  //how much does sorting even do here?
+
+  for (auto rball: correctedR){
+    for (auto lball: correctedL){
+      if (rball.Y == lball.Y){
+        //TODO (Kuba): Is there a method for judging whether the x values are close enough?
+        //TODO (Kuba): This assumes that the cameras are at the exact same height, which seems unlikely. We should do some sort of bounds/error correction here
+        pair<Mat, Mat> newPair;
+        newPair.first = lball;
+        newPair.second = rball;
+        pairedBalls.push_back(newPair);
+        //break; Assuming we can guaruntee there's only one match, break would optimize performance
```

Comment:
Yes, you do want a break
```suggestion
        break;
```